components/ntp_background_images/browser/ntp_background_images_service.cc

```diff
@@ -131,4 +496,76 @@ void NTPBackgroundImagesService::NotifyObservers() {
   }
 }
 
+void NTPBackgroundImagesService::OnPreferenceChanged(
+  const std::string& pref_name) {
+  DCHECK_EQ(kReferralPromoCode, pref_name);
+  const std::string new_referral_code = GetReferralPromoCode();
+  DVLOG(2) << __func__ << ": Got referral promo code: "
+                       << new_referral_code;
+  // If referral_code is not empty, it means referral code is fetched at first
+  // run or referral code is changed by another install.
+  if (!new_referral_code.empty()) {
+    const std::string previous_referral_code = GetCachedReferralPromoCode();
+    // Sync with referral code if it's not empty.
+    local_pref_->SetString(prefs::kNewTabPageCachedReferralPromoCode,
+                           new_referral_code);
+
+    // If previous referrer is super referrer and current referrer another
+    // super referrer, we should unregister previous one.
+    if (previous_referral_code != new_referral_code) {
+      DVLOG(2) << __func__ << ": Referrer was changed from "
+                           << previous_referral_code << " to "
+                           << new_referral_code;
+      UnRegisterSuperReferrerComponentIfRunning(previous_referral_code);
+    }
+
+    DetermineTargetComponent();
+  }
+}
+
+bool NTPBackgroundImagesService::LoadTestData() {
```

Comment:
this method name doesn't make sense to me. It may or may not load anything and it's not always test data. I think it makes more sense to just keep the contents of this method directly in `Init`

Reply:
Moved.